### PR TITLE
Add 'with javabridge.vm()' statement

### PR DESCRIPTION
**Pull request by manuels**
This adds javabridge.vm() which makes the usage of javabridge more 'pythonic' by using the `with` statement:

    with javabridge.vm(run_headless=True):
      print javabridge.run_script('java.lang.String.format("Hello, %s!", greetee);', 

(Not being able to use `with` has been bugging me for a while ;)

### CODE REVIEW

File: javabridge/tests/test_jutil.py [hunk 1]
```diff
@@ -661,6 +661,12 @@ def test_12_03_jref_create_and_lock(self):
             'javabridge.unlock_jref(ref_id)', d, d)
         javabridge.unlock_jref(ref_self)
         self.assertRaises(KeyError, javabridge.redeem_jref, ref_self)
+
+class TestJutil(unittest.TestCase):
+    def test_with_stmt(self):
+        with javabridge.vm(run_headless=True):
```
Comment (LeeKamentsky) [inline javabridge/tests/test_jutil.py]: I think this is going to fail, unless it's the only test that's run. Unfortunately, the VM can't be restarted or duplicated (It's a limitation of the JVM, not Javabridge), so this will either fail or cause other tests to fail.

Comment (manuels) [inline javabridge/tests/test_jutil.py]: So should I dump this unittest?

**Comment by LeeKamentsky** [comment]
Thanks, Manuels. It's not a bad idea, but the problem is that its use would have to be restricted to a top-level function (the equivalent of C's main() function) because it can only be used one time due to JVM restrictions on restarting.

If you want to use the javabridge in a thread, you have to call `javabridge.attach()` before executing any javabridge code and `javabridge.detach()` before exiting the thread. I've often thought that it would be nice to have a "with" statement to handle that:

    with javabridge.JB_Env() as env:
        # env is a JB_Env object, useful for low-level code. JB_Env

So I'll take the pull request if you remove the test you added - you can see how all the tests fail in Travis after the test you added runs.

**Comment by manuels** [comment]
Thanks, I must note that I have just used javabridge with bioformats and in these usecases I only had to use `start_vm/kill_vm` so I'm not really familiar with the `attach/detach` stuff.
